src/popover.ts

```diff
@@ -229,57 +160,28 @@ export function apply() {
     if (!(root instanceof ShadowRoot || root instanceof Document)) {
       return;
     }
-    const invoker = target.closest(popoverInvokerSelector);
-    let popoverTargetElement: HTMLElement | null = null;
-    if (
-      invoker instanceof HTMLButtonElement ||
-      invoker instanceof HTMLInputElement
-    ) {
-      if (invoker.popoverToggleTargetElement) {
-        popoverTargetElement = invoker.popoverToggleTargetElement;
-        if (popoverTargetElement) {
-          if (visibleElements.has(popoverTargetElement)) {
-            popoverTargetElement.hidePopover();
-          } else {
-            popoverTargetElement.showPopover();
-          }
-        }
-      } else if (invoker.popoverShowTargetElement) {
-        popoverTargetElement = invoker.popoverShowTargetElement;
-        if (
-          popoverTargetElement &&
-          !visibleElements.has(popoverTargetElement)
-        ) {
-          popoverTargetElement.showPopover();
-        }
-      } else if (invoker.popoverHideTargetElement) {
-        popoverTargetElement = invoker.popoverHideTargetElement;
-        if (popoverTargetElement && visibleElements.has(popoverTargetElement)) {
-          popoverTargetElement.hidePopover();
-        }
-      }
+    const invoker = target.closest('[popovertargetaction],[popovertarget]');
+    if (invoker) {
+      popoverTargetAttributeActivationBehavior(invoker as HTMLButtonElement);
+      return;
     }
-    hideOpenAutoPopovers(
-      popoverTargetElement ||
-        closestShadowPenetrating(openPopoverSelector, target),
-    );
   };
 
   const onKeydown = (event: Event) => {
     const key = (event as KeyboardEvent).key;
-    if (key === 'Escape' || key === 'Esc') {
-      hideOpenAutoPopovers();
+    const target = event.target as Element;
+    if (target && (key === 'Escape' || key === 'Esc')) {
+      hideAllPopoversUntil(target.ownerDocument, false, false);
```

Comment:
Is it intentional that this does not return focus to the previously-focused element?

Reply:
It's not. This should be `true`.

```suggestion
      hideAllPopoversUntil(target.ownerDocument, true, false);
```